File: LoginServer/core/models/protocol/getusers.go
package protocol

import (
	"app/core/models/database"
)

type GetUsersClient struct{
	SearchType		string
	SearchValue		string
}

func (o *GetUsersClient) Check() string{
	if (o.SearchType != "NoFilter" && o.SearchType != "") && o.SearchValue == ""{
		return "Sin valores de busqueda"
	}
	return ""
}

type UserInfo struct{
	Id				uint64
	Username 		string
	Name 			string
	Lastname 		string
	Documentid 		string
	Email 			string
	Cellphone 		string
	Permissions 	string
	Status			int
}

type GetUsersServer struct{
	ResponseCode	int
	ResponseString	string
	Info			[]UserInfo
}

func (o* GetUsersServer) AddItem(user database.Users){
	var item UserInfo
	
	item.Id = user.Id
	item.Username = user.Username
	item.Name = user.Name
	item.Lastname = user.Lastname
	item.Documentid = user.Documentid
	item.Email = user.Email
	item.Cellphone = user.Cellphone
	item.Permissions = user.Permissions
	item.Status = user.Status

	o.Info = append(o.Info,item)
}

File: cliente/src/services/Usuarios/GetUsers.js
import Configuration from '../../configuration/configuration';
import axios from 'axios';

class UserInfo {
    constructor(info){
        this.Id = info.Id
        this.Username = info.Username
        this.Name = info.Name
        this.Lastname = info.Lastname
        this.Cellphone = info.Cellphone
        this.Documentid = info.Documentid
        this.Email = info.Email
        //this.Permissions = info.Permissions
        this.Status = info.Status
    }
}

class GetUserObj {
    constructor(){
        this.Status = 0
        this.Message = ""
        this.BodyInfo = []
        this.BodyPosition = {}
        this.HeaderInfo = []
    }
}

export default class GetUser{
    async DoWork(searchtype,searchvalue) {
        var obj = {};
        var ret = new GetUserObj();
        var authorization = sessionStorage.getItem("authorization");
        
        obj["SearchType"] = searchtype;
        obj["SearchValue"] = searchvalue;

        try {
            const headers = {
                'Content-Type': 'application/json;charset=UTF-8',
                //'Access-Control-Allow-Headers': '*',
                'Authorization': 'Bearer ' + authorization,
            };

            const response = await axios.post(Configuration.API_LOGIN + "/GetUsers",JSON.stringify(obj),{ headers : headers});

            var data = response.data;

            ret.Message = data.Message != null ? data.Message : ""
            ret.Status = data.ResponseCode
            

            if (data.ResponseCode === 0 && data.Info !== null) {
                var pos = 0

                for (var info of data.Info) {
                    var user = new UserInfo(info)
                    ret.BodyInfo.push(user)
                    ret.BodyPosition[user.Id] = pos++
                }
            }

            return ret
        } catch (error) {
            ret.Message = error.message + " - Sin respuesta del servidor"
            ret.Status = 1

            return ret
        }
    }
}